peekingduck/pipeline/nodes/base.py

```diff
@@ -328,51 +384,74 @@ def _find_paths(self) -> Tuple[Path, Path]:
 
             if not weights_parent_dir.exists():
                 raise FileNotFoundError(
-                    f"The specified weights_parent_dir: {weights_parent_dir} does not exist."
+                    f"weights_parent_dir does not exist: {weights_parent_dir}"
                 )
             if not weights_parent_dir.is_absolute():
                 raise ValueError(
-                    f"The specified weights_parent_dir: {weights_parent_dir} "
-                    "must be an absolute path."
+                    f"weights_parent_dir must be an absolute path: {weights_parent_dir}"
                 )
 
-        weights_dir = weights_parent_dir / PEEKINGDUCK_WEIGHTS_SUBDIR
-        model_dir = weights_dir / self.config["weights"]["model_subdir"]
+        return (
+            weights_parent_dir
+            / PEEKINGDUCK_WEIGHTS_SUBDIR
+            / self.model_subdir
+            / self.config["model_format"]
+        )
 
-        return weights_dir, model_dir
+    def _get_weights_checksum(self) -> str:
+        with requests.get(f"{BASE_URL}/weights_checksums.json") as response:
+            checksums = response.json()
+        return checksums[self.model_subdir][self.config["model_format"]][
+            str(self.config["model_type"])
+        ]
 
-    @staticmethod
-    def _has_weights(weights_dir: Path, model_dir: Path) -> bool:
-        """Checks for model weight paths from weights folder.
+    def _has_weights(self, model_dir: Path) -> bool:
+        """Checks if the specified weights file is present in the model
+        sub-directory of the PeekingDuck weights directory.
 
         Args:
-            weights_dir (Path): Path to where all weights are stored.
-            model_dir (Path): Path to where weights for a model are stored.
+            model_dir (Path): /path/to/peekingduck_weights/<model_name> where
+                weights for a model are stored.
 
         Returns:
-            (bool): ``True`` if specified files/directories in
-            ``weights_dir`` exist, else ``False``.
+            (bool): ``True`` if specified weights file in ``model_dir``
+            exists and up-to-date/not corrupted, else ``False``.
         """
-        if not weights_dir.exists():
-            weights_dir.mkdir()
+        weights_path = model_dir / self.model_filename
+        if not weights_path.exists():
+            self.logger.warning("No weights detected.")
+            return False
+        if self.sha256sum(weights_path).hexdigest() != self._get_weights_checksum():
+            self.logger.warning("Weights file is corrupted/out-of-date.")
```

Comment:
when `self.sha256sum(weights_path).hexdigest() != self._get_weights_checksum()` evaluates to `True`, it will cause `_has_weights()` to return `False`. This will then prevent [this line](https://github.com/liyier90/PeekingDuck/blob/feat-download-individual-weights/peekingduck/pipeline/nodes/base.py#L320) in `download_weights()` from returning early and proceed with the download process.

The current behavior is to automatically re-download the weights when it's missing, corrupted, or out-of-date. No user intervention is required.